### PR TITLE
Preserve comments and elements order in Nuget.Config

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: https://github.com/dotnet/arcade/issues/3938

Completing the work that @mmitche started:

- Make `UpdatePackageSources` preserve comments already present in `NuGet.Config`
- Make `UpdatePackageSources` keep the order of the elements already present in `NuGet.Config`
- Add test infrastructure & test cases

### CODE REVIEW

**Comment by riarenas** [comment]
Should this be a hotfix to production?

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyFlowGraphOperation.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using Microsoft.DotNet.DarcLib.Helpers;
 using Microsoft.DotNet.Maestro.Client.Models;
 using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.Services.Invitation;
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyFlowGraphOperation.cs]: What is this?

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyFlowGraphOperation.cs [hunk 2]
```diff
@@ -64,6 +65,40 @@ public override async Task<int> ExecuteAsync()
                     flowGraph.PruneGraph(node => IsInterestingNode(targetChannel, node), edge => IsInterestingEdge(edge));
                 }
 
+                // For each node, determine number of changes that are made for a change starting at that node.
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyFlowGraphOperation.cs]: Are the changes to this file expected in this change? seem like something else

**Comment by riarenas** [comment]
Yeah I believe the base branch for this changes should be production. Inludes a bunch of changes already in that branch

**Comment by riarenas** [comment]
I'll create the PR to merge back production into master to get us the latest hotfixes, and that should make this diff make a littke more sense.

**Comment by riarenas** [comment]
You might want to rebase the changes in this branch against master so that the commits and changes included in the PR reflect the actual changes that are going to be going in.

**Comment by JohnTortugo** [comment]
Update:

- After the rebase + merge, etc. all unit tests are passing again. 
- I think the change to `GetDependencyFlowGraphOperation.cs` is unrelated and I didn't find that change in master or production. @mmitche is that a complete patch / should I keep it?

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyFlowGraphOperation.cs (hunk 2 above)
Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyFlowGraphOperation.cs]: Oh interesting...I wonder how this got in here.  Remove.  It was related to some calculation I was doing

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs [hunk 3]
```diff
@@ -275,34 +281,93 @@ private XmlDocument UpdatePackageSources(XmlDocument nugetConfig, HashSet<string
...
+                return;
+            }
+
+            if (packageSourcesNode.FirstChild.NodeType != XmlNodeType.Element
+                || !packageSourcesNode.FirstChild.Name.Equals("clear", StringComparison.OrdinalIgnoreCase))
+            {
+                SetElement(nugetConfig, packageSourcesNode, VersionFiles.ClearElement);
+            }
+
+            // Add beginning comment
```
Comment (mmitche) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs]: Nit: This is the end comment.

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyFlowGraphOperation.cs (hunk 2 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyFlowGraphOperation.cs]: Removed.

**Comment by JohnTortugo** [comment]
Solved the nits. Wait for CI to finish.

**Pull request merged**